src/lib.rs

```diff
@@ -110,39 +115,34 @@ fn random_gen_dust_concentration() -> String {
     rng.gen_range(0..=1000).to_string()
 }
 
-pub fn run(cli: &Cli) {
-    println!("\r\ncli: {:?}\r\n", cli);
-
+fn send_request(url: &str, client: Client, debug: bool) {
     let dust_concentration = random_gen_dust_concentration();
     let air_purity = AirPurity::from_value(dust_concentration.parse::<u16>().unwrap()).to_string();
     let reading = Reading::new(
         random_gen_temperature(),
         random_gen_humidity(),
         random_gen_pressure(),
         dust_concentration,
-        air_purity
+        air_purity,
     );
 
     let json = serde_json::to_string(&reading).unwrap();
-    const URL: &str = "http://localhost:4000/api/readings/add";
-
-    println!("Sending POST request to {} as JSON: {}", URL, json);
-
-    let client = Client::new();
+    println!("Sending POST request to {} as JSON: {}", url, json);
     let res = client
-        .post(URL)
+        .post(url)
         .header(CONTENT_TYPE, "application/json")
         .body(json)
         .send();
     match res {
-        Ok(response) => {
-            match cli.debug {
-                true => println!("Response from Ambi backend: {:#?}", response),
-                false => println!("Response from Ambi backend: {:?}", response.status().as_str())
-            }
-        }
+        Ok(response) => match debug {
```

Comment:
I think I prefer to use an if statement here if debug is a bool

Reply:
Any particular reason why or just a syntactic preference that reads more clearly?

Reply:
I just checked what clippy says about it, it doesn't complain about using match with a bool. So I guess it's just preference. I certainly find it more readable to read using an if statement, but YMMV. I won't insist.

Reply:
Cool, Caleb was reusing a pattern that I put in place which I originally got from the `clap` project itself in an example. We're all new to Rust, so it's actually quite a useful conversation to have on what makes for strong Rust and what might just be individual preference.